src/manager/k8s.service.ts

```diff
@@ -1,24 +1,193 @@
 import { Injectable } from '@nestjs/common';
 import { IAgentManager } from './agent.manager.interface';
 import { AgentConfig } from './agent.config';
+import { KubeConfig, CoreV1Api, V1Secret, V1Deployment, V1Pod, V1Service, V1PodCondition } from '@kubernetes/client-node';
+import { Logger } from 'protocol-common/logger';
+import { readFile } from 'fs';
+import cryptoRandomString from 'crypto-random-string';
 
 /**
  * Starts and stops agents within a kubernetes environment
  */
 @Injectable()
 export class K8sService implements IAgentManager {
 
+    // newKubeClient
+    // This configuration is only available if you run the server on a filesystem provisioned with a k8s client.
+    // When running in k8s, this config will be available via the configured k8s service account of the pod.
+    // Locally, this config is provided via ~/.kube/config.
+    newKubeClient = (): CoreV1Api => {
+        const kc = new KubeConfig();
```

Comment:
since this is typescript we can make our functions like this:  https://github.com/kiva/aries-guardianship-agency/blob/1ddeb9b390bedc9858ff9f5ee895cf7cb2ce4516/src/manager/docker.service.ts#L27. (leave off aysnc if function isnt async.  Lets change all of the functions to match that.